lib/passport-saml/saml.js

```diff
@@ -661,9 +661,14 @@ SAML.prototype.validatePostResponse = function (container, callback) {
   })
   .then(() => this.certsToCheck())
   .then(certs => {
-    // Check if this document has a valid top-level signature
+    // Check if this document has a valid top-level signature which applies to the entire XML document
     var validSignature = false;
-    if (this.options.cert && this.validateSignature(xml, doc.documentElement, certs)) {
+    if (this.options.cert &&
+      this.validateSignature(xml, doc.documentElement, certs) &&
+      Array.from(doc.childNodes).filter(
+          (n) => n.tagName != null && n.childNodes != null
```

Comment:
Usually I'd suggest changing `!=` to `!==` to avoid type coercion surprises, but if this is copied directly from upstream it's probably better to leave it as is.